### PR TITLE
Reader questions

### PR DESCRIPTION
**Pull request by regiskuckaertz**
Takes the code from frontend with a few changes:
- Ben's new design
- No feature switch yet, so this version only exposes the 404 test, which is 100% anyway. That's ok, we can update when the email stuff is ready to go.
- No business logic, that must be handled upstream (`isClosed` or AMP)
- The value sent if the user chooses "Notification" is now `notification`. Whether the event is coming from the app or frontend can be detected via the platform field.

- [x] Test on frontend
- [x] Test in apps

A couple of screengrabs:
screen shot 2017-11-23 at 10 24 28
screen shot 2017-11-23 at 11 03 22
screen shot 2017-11-23 at 11 03 35
screen shot 2017-11-23 at 11 04 04
screen shot 2017-11-23 at 11 04 28

### CODE REVIEW

**Comment by mchv** [comment]
> The value sent if the user chooses "Notification" is now notification. Whether the event is coming from the app or frontend can be detected via the platform field.

We need to display `app notification` and `web notification` in each case, so why not having a parameter in the context that say that is called in apps and then send the same notification as before?

**Comment by mchv** [comment]
@regiskuckaertz exactly like in https://github.com/guardian/frontend/pull/18213/files the js sides need to be able to take a parameter about the context (in this case in app).

**Comment by regiskuckaertz** [comment]
We could do that, but then it allows anyone to fork the code and this is precisely what we're distancing ourselves from. I'd like to keep it that way as long as possible, and in this case the information is already available and so can be retrieved in Ophan.

File: src/main/resources/storyquestions/article/index.fjs [hunk 1]
```diff
@@ -1 +1,164 @@
...
\ No newline at end of file
+//@flow
+
+import { Actions, ComponentTypes } from 'lib/ophan';
+import { fromEvent } from 'lib/events';
+import type { Channel } from 'channels';
+
+type QuestionId = string;
+type QuestionText = string;
+type Question = [?QuestionId, ?QuestionText];
```
Comment (tomrf1) [inline src/main/resources/storyquestions/article/index.fjs]: why did you choose an array here, rather than an object?

Comment (regiskuckaertz) [reply src/main/resources/storyquestions/article/index.fjs]: Flow has support for "tuples" as lists of n-items: https://flow.org/en/docs/types/tuples/

**Comment by mchv** [comment]
> We could do that, but then it allows anyone to fork the code and this is precisely what we're distancing ourselves from. I'd like to keep it that way as long as possible, and in this case the information is already available and so can be retrieved in Ophan
@regiskuckaertz I don't agree, we want to be able to have some customisation between context. In this case this is painful to not have the expected name displayed in the app, and then have to do post-treatment on the datalake to exploit the data, especially as we already have data with `app-notification` and `web-notification`

**Comment by regiskuckaertz** [comment]
Yeah of course, there is customization but it must happen at the platform-level. The idiomatic way is via CSS variables for CSS and via services for JS. So in that case, I might add a `ContextService` service that exposes a method `getOphanCodeFor(id)` that returns `notification-web` or `notification-app`.

**Comment by regiskuckaertz** [comment]
[Atom library: Add reader questions to library](https://trello.com/c/ZqjKZvo2/196-atom-library-add-reader-questions-to-library)

**Comment by regiskuckaertz** [comment]
[Make Reader Question atom look like an atom](https://trello.com/c/bANkC5VM/182-make-reader-question-atom-look-like-an-atom)

**Comment by regiskuckaertz** [comment]
@chriswilk was fine with a common label in this case, so I'll go with this. I'm not shying away from contextual customization, but it is a feature that is complex enough to warrant its own feature branch. I don't want to shoehorn a quick fix that will end up making the whole library vulnerable to a lack of discipline in the developer's head, i.e. an `inApp` flag would allow someone to write:
```
@if(inApp) {
  render atom for app
} else {
  render atom for web
}
```
That should not be allowed. Any customization must be provided at the calling site and be out of reach from within the library.

**Comment by regiskuckaertz** [comment]
I think this customization will probably happen in Scala land, with the benefit of being verified by the compiler (so we don't risk null exceptions) but also because when the lib is capable of rendering AMP, we won't be able to use the stuff defined in JS.

**Pull request merged**